sympy/series/formal.py

```diff
@@ -1464,6 +1452,130 @@ def __rmul__(self, other):
         return self.__mul__(other)
 
 
+class FiniteFormalPowerSeries(FormalPowerSeries):
+    """Represents Finite Formal Power Series of a function.
+
+    No computation is performed. This class should only to be used to represent
+    a series. No checks are performed.
+
+    The difference between `FiniteFormalPowerSeries` and `FormalPowerSeries` is that
+    the coefficient sequence of a `FiniteFormalPowerSeries` is a list of coefficients,
+    instead of a sequence as in a `FormalPowerSeries` object.
+
+    `product`, `compose`, and `inverse` functions of `FormalPowerSeries` class, return
+    an object of this class.
+
+    See Also
+    ========
+
+    sympy.series.formal.FormalPowerSeries
+    sympy.series.formal.fps
+    """
+
+    def __init__(self, *args):
+        pass
+
+    @property
+    def infinite(self):
+        raise NotImplementedError("No infinite version for an object of"
+                     " FiniteFormalPowerSeries class.")
+
+    def _eval_terms(self, n):
+        pass
+
+    def _eval_term(self, pt):
+        raise NotImplementedError
+
+    def polynomial(self, n):
+        return self._eval_terms(n)
+
+    def _eval_derivative(self, x):
+        raise NotImplementedError
+
+    def integrate(self, x):
+        raise NotImplementedError
+
+
+class FormalPowerSeriesProduct(FiniteFormalPowerSeries):
+    def __new__(cls, *args):
+        args = map(sympify, args)
+        return FormalPowerSeries.__new__(cls, *args)
+
+    @property
+    def f(self):
+        return self.args[0][0]
+
+    @property
+    def g(self):
+        return self.args[0][1]
+
+    def _eval_terms(self, n):
+        f, g = self.f, self.g
+        coeff1, coeff2 = self.args[4][0][0], self.args[4][0][1]
+
+        aks = convolution(coeff1[:n], coeff2[:n])
+
+        terms = []
+        for i in range(0, n):
+            terms.append(aks[i] * self.xk.coeff(i))
+
+        return Add(*terms)
+
+
+class FormalPowerSeriesCompose(FiniteFormalPowerSeries):
+    def __new__(cls, *args):
+        args = map(sympify, args)
+        return FormalPowerSeries.__new__(cls, *args)
+
+    def __init__(self, *args):
+        k = self.xk.variables[0]
+        self.fact_seq = sequence(factorial(k), (k, 1, oo))
+
+    @property
+    def outer_func(self):
+        return self.args[0][0]
+
+    @property
+    def inner_func(self):
+        return self.args[0][1]
+
+    def _eval_terms(self, n):
+        f, g = self.outer_func, self.inner_func
+        ffps, gfps = fps(f), fps(g)
+
+        terms = [ffps._eval_term(0)]
+
+        for i in range(1, n):
+            bell_seq = gfps.coeff_bell(i)
+            seq = (self.ak * bell_seq)
+            terms.append(Add(*(seq[:i])) / self.fact_seq[i-1] * self.xk.coeff(i))
+
+        return Add(*terms)
+
+
+class FormalPowerSeriesInverse(FiniteFormalPowerSeries):
+    def __new__(cls, *args):
+        args = map(sympify, args)
+        return FormalPowerSeries.__new__(cls, *args)
```

Comment:
I think you don't need it, since your are using ``FormalPowerSeries`` new anyway.